File: Programmers/Lv_0/겹치는선분의길이.cpp
#include <iostream>
#include <string>
#include <vector>
using namespace std;

int arr[200];

int solution(vector<vector<int>> lines) {
    int answer = 0;

    for (int i = 0; i < lines.size(); i++) {
        for (int j = lines[i][0]; j < lines[i][1]; j++) {
            arr[j+100]++;
        }
    }

    for (int i = 0; i < 200; i++) {
        if (arr[i] >= 2) answer++;
        // cout << arr[i] << " ";
    }

    return answer;
}

int main() {
    ios_base::sync_with_stdio(0);
    cin.tie(0);

    int data;
    vector<int> v;
    vector<vector<int>> lines;

    for (int i = 0; i < 3; i++) {
        for (int j = 0; j < 2; j++) {
            cin >> data;
            v.push_back(data);
        }
        lines.push_back(v);
        v.clear();
    }

    // cout << '\n';
    // for (int i = 0; i < 3; i++) {
    //     for (int j = 0; j < 2; j++) {
    //         cout << (*(lines.begin() + i))[j] << ' ';
    //     }
    //     cout << '\n';
    // }

    cout << solution(lines);
    
    return 0;
}

File: Programmers/Lv_0/겹치는선분의길이.js
function solution(lines) {
    var answer = 0;
    let lineMap = new Array(200); // 선분들이 놓일 공간
    lineMap.fill(0);

    for (let i = 0; i < 3; i++) {
        let left = lines[i][0];
        let right = lines[i][1];

        for (let j = left; j < right; j++) {
            lineMap[j+100] += 1;
        }
    }

    for (let i in lineMap) {
        if (lineMap[i] > i) {
            answer += 1;
        }
    }

    return answer;
}